concat.hpp

```diff
@@ -0,0 +1,119 @@
+#include "utils.hpp"
+
+#include <hls_stream.h>
+#include <ap_int.h>
+#include <algorithm>
+
+
+template<typename Tsrc, typename Tdst>
+void convert_vector(Tsrc& src, Tdst& dst){
```

Comment:
If in this header file, this is a _local_ utility. Move into the anonymous namespace below.
Also, request `#pragma HLS inline`.
If only used for `hls::vector`, make this explicit in the signature and use this as an opportunity to enforce that `src` and `dst` are of the same size. Use the _inferrable_ vector length parameter as the loop bound below.